flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/debezium/reader/BinlogSplitReaderTest.java

```diff
@@ -279,56 +287,130 @@ public void testReadBinlogFromLatestOffset() throws Exception {
                     "+I[2003, user_24, Shanghai, 123567891234]"
                 };
         List<String> actual =
-                readBinlogSplitsFromLatestOffset(dataType, sourceConfig, expected.length);
+                readBinlogSplitsFromLatestOffset(
+                        createBinlogReader(sourceConfig), dataType, sourceConfig, expected.length);
         assertEqualsInOrder(Arrays.asList(expected), actual);
     }
 
-    private List<String> readBinlogSplitsFromLatestOffset(
-            DataType dataType, MySqlSourceConfig sourceConfig, int expectedSize) throws Exception {
-        final StatefulTaskContext statefulTaskContext =
-                new StatefulTaskContext(sourceConfig, binaryLogClient, mySqlConnection);
+    @Test
+    public void testHeartbeatEvent() throws Exception {
```

Comment:
Please also test SnapshotSplitReader and I think it will fail as `RecordUtils#normalizedSplitRecords` doesn't consider the nee introduced heartbeat event type. A solid way is that we should disable heartbeat event for snapshot split.